hedera-node/src/main/java/com/hedera/services/contracts/execution/EvmTxProcessor.java

```diff
@@ -366,15 +352,17 @@ protected TransactionProcessingResult execute(
                     gasPrice,
                     initialFrame.getOutputData(),
                     mirrorReceiver,
-                    stateChanges);
+                    stateChanges,
+                    hederaTracer.getActions());
```

Comment:
Reading this I think dagger injection of the tracer is wrong.  It's an implementation detail and not used across components.  With that we can then pass in the actions object to it's constructor and not need to query it from an interface.